OpenSim/Tools/InverseKinematicsTool.cpp

```diff
@@ -329,6 +329,10 @@ bool InverseKinematicsTool::run()
                 int worst = -1;
 
                 ikSolver.computeCurrentSquaredMarkerErrors(squaredMarkerErrors);
+                // if a marker is not present in the model or its reference value 
+                // is NaN it is not included. Therefore, size of the errors can
+                // vary from frame to frame.
```

Comment:
Perhaps an Issue should be opened about this? If `squaredMarkerErrors` changes in size, then it seems like the caller would have trouble monitoring the error of a particular marker (its index in the vector will be changing). Furthermore, I'm not sure `getMarkerNameForIndex()` below (on L353) will report the correct marker name if a marker with a lower index has dropped out. I would expect `computeCurrentSquaredMarkerErrors()` to always return a vector of the same (max) size; the caller would then step through (as on L336–342) and ignore any `NaN`s.